payment/src/main/java/org/killbill/billing/payment/bus/PaymentBusEventHandler.java

```diff
@@ -101,6 +115,8 @@ public void processInvoiceEvent(final InvoiceCreationInternalEvent event) {
                                                          properties, paymentControlPluginNames, callContext, internalContext);
         } catch (final AccountApiException e) {
             log.error("Failed to process invoice payment", e);
+        } catch (final InvoiceApiException e) {
+            log.error("Failed to process invoice payment", e);
         } catch (final PaymentApiException e) {
             // Log as error unless:
             if (e.getCode() != ErrorCode.PAYMENT_NULL_INVOICE.getCode() /* Nothing left to be paid */ &&
```

Comment:
@matias-aguero-hs All the changes in that file should be reverted; we don't want that handler to know anything about invoice.

Instead the check out the invoice status should be made into the [InvoicePaymentControlPluginApi](https://github.com/killbill/killbill/blob/master/payment/src/main/java/org/killbill/billing/payment/invoice/InvoicePaymentControlPluginApi.java#L255), similarly to what we do when we check if the balance is positive.